runtime/src/main/java/org/corfudb/runtime/collections/CorfuTable.java

```diff
@@ -228,13 +227,13 @@ public V get(@ConflictParameter Object key) {
      *
      * @param indexName Name of the the secondary index to query.
      * @param indexKey  The index key used to query the secondary index
-     * @return A collection of Map.Entry<K, V>
+     * @return A stream of {@code Map.Entry<K, V>}
      */
     @SuppressWarnings("unchecked")
     @Accessor
     public @Nonnull
     <I>
-    Collection<Entry<K, V>> getByIndex(@Nonnull Index.Name indexName, I indexKey) {
+    Stream<Entry<K, V>> getByIndex(@Nonnull Index.Name indexName, I indexKey) {
```

Comment:
This is a non-backwards compatible change. I am guessing the upper clients are not consuming this API directly?

Reply:
There are a few instances in NSX where this is consumed directly, namely in BaseIdentifiableObjectDao, but they too, turn it into a stream immediately. The call to stream can be removed in these instances when pushing the artifact update.

Reply:
Actually, in keeper, this is unused code; So I don't think it should be an issue.